backend/open_webui/utils/user_encryption.py

```diff
@@ -0,0 +1,373 @@
+        
+        # Convert dict to JSON string
+        json_str = json.dumps(data, separators=(',', ':'), ensure_ascii=False)
+        plaintext = json_str.encode('utf-8')
+        
+        # Create AESGCM instance and encrypt
+        aesgcm = AESGCM(key)
+        nonce = secrets.token_bytes(12)
+        ciphertext = aesgcm.encrypt(nonce, plaintext, None)
+        
+        # Combine nonce + ciphertext and return base64 encoded
+        encrypted_data = nonce + ciphertext
+        return base64.b64encode(encrypted_data).decode('utf-8')
+    
+    def decrypt_data(self, encrypted_data: str, user_id: str) -> dict:
+        """Decrypt data using AES-256-GCM with the user's encryption key."""
+        # Handle empty data
+        if not encrypted_data or not isinstance(encrypted_data, str):
+            return {}
+        
+        encrypted_data = encrypted_data.strip()
+        if not encrypted_data:
+            return {}
+        
+        # Check if data is plain JSON (backward compatibility)
+        if self._is_plain_json(encrypted_data):
+            try:
+                return json.loads(encrypted_data)
+            except json.JSONDecodeError:
+                return {}
+        
+        # Decrypt AES-256-GCM data
+        try:
+            # Get user's encryption key
+            key = self._get_user_encryption_key(user_id)
+            
+            # Decode base64
+            encrypted_bytes = base64.b64decode(encrypted_data.encode('utf-8'))
+            
+            # Extract nonce and ciphertext (nonce is first 12 bytes)
+            if len(encrypted_bytes) < 12:
+                return {}
+                
+            nonce = encrypted_bytes[:12]
+            ciphertext = encrypted_bytes[12:]
+            
+            # Decrypt and parse JSON
+            aesgcm = AESGCM(key)
+            decrypted_bytes = aesgcm.decrypt(nonce, ciphertext, None)
+            return json.loads(decrypted_bytes.decode('utf-8'))
+            
+        except Exception:
+            # Graceful degradation: return empty dict instead of crashing
+            return {}
+    
+    def _is_plain_json(self, data: str) -> bool:
+        """Check if data appears to be plain JSON (not encrypted)."""
+        if not data:
+            return False
+        stripped = data.strip()
+        return stripped.startswith('{') and stripped.endswith('}')
+    
+    def is_encrypted(self, data: str) -> bool:
+        """Check if data appears to be encrypted (not plain JSON)."""
+        return not self._is_plain_json(data)
+    
+    def _is_base64_encoded(self, data: str) -> bool:
+        """Check if data appears to be base64 encoded."""
+        if not data or not isinstance(data, str):
+            return False
+        
+        data = data.strip()
+        if not data:
+            return False
+        
+        # Base64 pattern check (letters, numbers, +, /, = padding)
+        base64_pattern = re.compile(r'^[A-Za-z0-9+/]*={0,2}$')
```

Comment:
Do we need to implement the base64 encoding check by ourselves? Should we simply rely on `base64.b64decode` for this?